src/Native/Unix/System.Native/pal_random.c

```diff
@@ -2,109 +2,206 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#include <stdlib.h>
-#include <stdint.h>
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
 #include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
 #include <sys/stat.h>
-#include <fcntl.h>
-#include <assert.h>
-#include <unistd.h>
 #include <time.h>
-#include <errno.h>
+#include <unistd.h>
 
 #include "pal_config.h"
 #include "pal_random.h"
+
+#if HAVE_SYS_GETRANDOM
+#include <sys/syscall.h>
+
+#ifndef GRND_NONBLOCK
+#define GRND_NONBLOCK 0x1
+#endif // GRND_NONBLOCK
```

Comment:
This definition can be replaced in favor of system header. `GRND_NONBLOCK` requires `#include <sys/random.h>` or `<linux/random.h>`; the former one is preferred to make it work on Solaris as well: https://docs.oracle.com/cd/E88353_01/html/E37841/getentropy-2.html (tested with clang39 and gcc7 on Ubuntu, and gcc7 on SmartOS).

Reply:
Done. Also checking for the existence of `sys/random.h` in `configure.cmake`, just in case.